### ISSUE TITLE
Tolgee doesn't parse ruby yaml correctly for some cases

### ISSUE THREAD

**Issue by cccccroge** [tolgee/tolgee-platform#2574]
**Describe the bug**
In our Rails project, some of the values result in extra single quote (`'`) when I check on Tolgee admin.

Here's the cases we found:
1. upload `key: value 'value'` to tolgee result in `value 'value''` (there's an extra single quote in the end)
2. upload `key: value'%{variable}'` to tolgee result in `value''<variable>''` (there are two single quotes around variable, which should be one)
3. upload `key: value{{variable}}value` to tolgee result in `value'{{'variable'}}'value` (two single quotes around {{ and }} shouldn't exist)
4. upload `key: "{variable} value"` to tolgee result in `'{'file'}' value` (two single quotes around { and } shouldn't exist)

**To Reproduce**
1. Create tolgee config that use Ruby format:
```
{
  "$schema": "https://tolgee.io/cli-schema.json",
  "format": "YAML_RUBY",
  "push": {
    "files": [
       // these files should contain the patterns mentioned above
    ],
    "forceMode": "OVERRIDE",
    "convertPlaceholdersToIcu": false
  }
}
```
2. Push to the target url
```
npx @tolgee/cli@2.1.4 push \
      --config ./test.tolgeerc.json \
      --api-url ${TOLGEE_INSTANCE_URL} \
      --api-key ${TOLGEE_PERSONAL_ACCESS_KEY} \
      --project-id ${TOLGEE_PROJECT_ID_RAILS} \
      --verbose
```
3. Go to translations page in tolgee host
4. It shows unexpected result
e.g.
<img width="1032" alt="image" src="https://github.com/user-attachments/assets/b82e765a-6797-43d3-b0b6-ebc659b95537">

**Expected behavior**
I expect tolgee don't transform value that is not interpolated string (`%{xxx}`) as they are valid yaml values. e.g. single quote (`'`) and curly braces (`{`, `}`, `{{`, `}}`) should not be escaped.

**Screenshots**
see above

**Versions and environment**
- Tolgee Platform version: 3.68.2
- Environment: local docker (https://hub.docker.com/layers/tolgee/tolgee/v3.68.2/images/sha256-cd8a247cba511ed846cbc51a36b221c24b69a18b32cae63ed3db36f3a3e87a06?context=explore) 
- Browser: not relevant

**Additional context**

**Comment by Anty0** [comment]
Hi! Thank you for the report.

When importing translations to tolgee, all strings are converted to ICU format by default (can be disabled). As part of this conversion, all characters with a special meaning in the ICU format have to be escaped, which is what you see now. You can read more [here](https://tolgee.io/platform/translation_process/tolgee_universal_icu_placeholders#escaping).

You can try to export your translations immediately after importing them. The resulting exported file should look the same as the imported one since all your strings will be converted to the selected format.

When needed, you can disable ICU conversion for the whole project (more info [here](https://tolgee.io/platform/translation_process/tolgee_universal_icu_placeholders#disabling-support-for-universal-icu-placeholders-globally)), but this will also disable support for variables and other ICU features for that project.

**Comment by Anty0** [comment]
@cccccroge Feel free to reopen the issue if there is something I've missed.

**Comment by Anty0** [comment]
@cccccroge Thank you for the clarification! I was able to reproduce the issue, and it should be fixed once #2629 gets released.
